pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -549,11 +603,47 @@ func (r *ReconcileIntrusionDetection) setDefaultsOnIntrusionDetection(ctx contex
 				},
 			},
 		}
+	}
 
-		if err := r.client.Update(ctx, ids); err != nil {
-			return err
+	if !isManagedCluster { // does not set defaults for managed clustes the AD Fields specs will simply be ignored
+		if len(ids.Spec.AnomalyDetection.StorageClassName) > 0 {
+			// User specified a storage class, set to persistent
+			ids.Spec.AnomalyDetection = operatorv1.AnomalyDetectionSpec{
+				StorageType:      operatorv1.PersistentStorageType,
```

Comment:
We shouldn't overwrite this if it is already set. That of course can lead to an invalid config with a storageClassName and Ephemeral but that should be caught by validation and reported to the user through a SetDegraded and blocking changes until that is fixed.